plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -152,21 +155,49 @@ func (c *MaxPDVolumeCountChecker) filterVolumes(volumes []api.Volume, namespace
 		} else if vol.PersistentVolumeClaim != nil {
 			pvcName := vol.PersistentVolumeClaim.ClaimName
 			if pvcName == "" {
-				return fmt.Errorf("PersistentVolumeClaim had no name: %q", pvcName)
+				// if the PVC does not have a name, log the error
+				// and count the PV towards the PV limit
+				// generate a random volume ID since its required for de-dup
+				glog.Errorf("PersistentVolumeClaim with no name found in namespace: %q", namespace)
+				source := rand.NewSource(time.Now().UnixNano())
+				generatedID := "noClaimName" + strconv.Itoa(rand.New(source).Intn(1000000))
+				filteredVolumes[generatedID] = true
+				return nil
```

Comment:
API validation requires the PVCVolumeSource to have a ClaimName.  This condition shouldn't ever happen.  

Should the pod be able to be scheduled in the first place? It will fail at mount time.